### DISCUSSION TITLE
Disable swipe gesture on TabBar

### DISCUSSION THREAD

**Discussion by astanb** [grahammendick/navigation#591] [Q&A] [answered]
I'd like to be able to disable the swipe gesture on the `TabBar` on Android. My use case is that I have a list of swipeable components in the tab and can't swipe them because the TabBar is receiving the gesture. Can it be done?

**Comment by grahammendick** [comment, marked as answer]
The `TabBar` on Android maps to the [ViewPager](https://developer.android.com/reference/kotlin/androidx/viewpager/widget/ViewPager). This widget does support horizontally scrolling content. I know this because [I investigated it when I added right-to-left (RTL) language support](https://github.com/grahammendick/navigation/pull/445). Android, instead of adding RTL support to the ViewPager, decided to write a whole new widget called [ViewPager2](https://developer.android.com/jetpack/androidx/releases/viewpager2). I was going to move over to ViewPager2 but found out that it doesn’t support horizontally scrolling content. I didn’t want to lose this functionality, so the `TabBar` maps to the ViewPager for LTR and ViewPager2 for RTL languages. 

So you should be able to add a horizontal ScrollView to a tab in the `TabBar`, for example.

**Reply by astanb** [reply]
You're right, `ScrollView` does indeed work ok and intercepts the gesture ahead of the tab bar. I'm actually using `PanGestureHandler` from react-native-gesture-handler for my swiper and that doesn't seem to be able to override the pager yet. 

Looks like it's possible to disable scrolling at the native level, see https://stackoverflow.com/questions/7814017/is-it-possible-to-disable-scrolling-on-a-viewpager

Would you accept a PR that implements it?

**Reply by grahammendick** [reply]
What happens if you put the `PanGestureHandler` inside a horizontal `ScrollView`? Who gets the swipe?

**Reply by astanb** [reply]
Tried that in various permutations, I can stop the ViewPager from scrolling and get the onStart for the swiper but not the onActive or onEnd events. Going to dig into it and see if there's something else I'm missing. I tested with putting a [slider](https://www.npmjs.com/package/@react-native-community/slider) in the tab and that works fine (doesn't trigger the ViewPager to move) so there must be something going on...

**Reply by grahammendick** [reply]
I meant for you to try it without a TabBar. Just a PanGestureHandler inside a plain old ScrollView. Was wondering who won the swipe war in that case

**Reply by astanb** [reply]
Fixed it! Dug into my code and found that I had `activeOffsetX` set to [-50, 50] (this swiper is a in a vertically scrolling long list of items). Setting it to [-10, 10] still allows vertical scroll but most importantly I can now swipe left and right and the ViewPager doesn't get the gesture. So it looks like ViewPager must have a tolerance of somewhere between 10 and 50 before it takes over the pan event.
